src/GNDStk.hpp

```diff
@@ -81,5 +82,36 @@ namespace GNDStk {
 // xml/json/tree conversions
 #include "GNDStk/convert.hpp"
 
+
+// ------------------------
+// GNDStk "core interface"
+// ------------------------
+
+/*
+In a user code, writing this:
+
+   using namespace njoy::GNDStk::core;
+
+gives the same effect as writing this:
+
+   using namespace njoy::GNDStk;
+   using namespace njoy::GNDStk::basic;
+
+Meaning: We consider our "core interface" to consist of everything in GNDStk::
+proper, plus our basic:: set of meta_t and child_t keywords. Those are the
+meta_t and child_t objects with <void> type, so that they return metadata and
+nodes in their original tree form. Note that basic:: itself brings in meta_t
+and child_t objects from its own nested meta:: and child:: namespaces. Those
+are separate because there's a small amount of overlap between allowable GNDS
+metadatum names and node names. With this scheme, if a name you wish to use
+isn't one of the overlapping names, just use it. If it is, then prefix with
+meta:: or child:: depending on which one you want.
+*/
+
+namespace core {
+   using namespace GNDStk;
```

Comment:
We will (soon) have the standardised versioned namespaces in the GNDStk namespace as well, so we may want to move the code that is currently in the GNDStk namespace to the core namespace and it's own core.hpp file.